### DISCUSSION TITLE
Metadata and Static Files location in Nextjs 13 App Router: Right way /app or /public?

### DISCUSSION THREAD

**Discussion by gurbaaz27** [vercel/next.js#50593] [App Router] [answered]
I am using Next.js `13.4.2`, and I have 2 doubts, pertaining to the location of static files and the generation of metadata

1. The [link](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/app-icons) suggests keeping app icons in the `/app` directory, but I thought we usually keep this stuff(images, robots.txt, etc.) in `/public`  directory. So which one is the right way? And also, now should I keep my `robots.txt` in `/app` or `/public`?
2. Before tha App router was stable and docs were in beta, I followed them to generate metadata as follows

```typescript
export const metadata = {
  title: 'my-title',
  description: 'here-went-the-description',
  category: 'education',
  keywords: ['a', 'b', 'c'],
  openGraph: {
    title: 'my-title',
    description: 'here-went-the-description',
    images: ['/logo.svg', '/logo-square.svg'],
    url: 'https://www.url.com',
    siteName: 'sitename',
  },
  metadataBase: new URL('https://www.metabase.com'),
  icons: {
    icon: '/favicon.ico',
    apple: '/apple-icon.jpg'
  }
}
```

But the new docs seem to suggest that instead of stating image paths here as I did, I simply need to keep them in `/app` segment, as mentioned [here](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/app-icons) and [here](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/opengraph-image). So again, which one is the right way?

Glad if someone could help.

**Comment by devjiwonchoi** [comment, marked as answer]
Using the `/public` folder is [optional](https://nextjs.org/docs/getting-started/installation#create-the-public-folder).
> You can optionally create a public folder to store static assets such as images, ...

If you use `/public` folder, you can put your `favicon.ico` inside the `/public` folder.

The reason why [docs](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/app-icons#apple-icon) states to set the icon in the root `/app` segment
> App icons can only be set in the root /app segment.

is because the `/public` folder is optional, therefore IF YOU CHOSE NOT TO USE `/public` folder, 
you should place the icon on the **root** `/app` segment.

Either choice you can use `/` for the icon path.

# UPDATE

## Where to add the `robots.txt`?

The `robots.txt` is also a [Static Asset](https://nextjs.org/docs/app/building-your-application/optimizing/static-assets) that goes along with the `app icons`, which means to place on either `/public` or root `/app` folder.

> This folder is also useful for robots.txt, favicon.ico, Google Site Verification, and any other static files (including .HTML)!

## Where to set the Metadata?

The same goes for [Metadata Files](https://nextjs.org/docs/app/api-reference/file-conventions/metadata), but note that when using `Metadata API`, it is recommended to define `metadata` inside `layout.js` or `page.js`.

> To define static metadata, export a [Metadata object](https://nextjs.org/docs/app/api-reference/functions/generate-metadata#metadata-object) from a layout.js or static page.js file.

## Reference
- App Router Metadata `robots.txt`: https://nextjs.org/docs/app/api-reference/file-conventions/metadata/robots
- App Router Metadata: https://nextjs.org/docs/app/building-your-application/optimizing/metadata
- App Router Dynamic Metadata: https://nextjs.org/docs/app/building-your-application/optimizing/metadata#dynamic-metadata

**Reply by gurbaaz27** [reply]
Thanks a lot! And any comments on the second doubt?

**Reply by devjiwonchoi** [reply]
@gurbaaz27 Apologies for missing out on your other questions. Updated the previous answer.

**Reply by gurbaaz27** [reply]
Thanks for the prompt replies! Worked for me.
